### PR TITLE
feat(table/web) : Add table component

### PR DESCRIPTION
**Pull request by pragyajha**

### CODE REVIEW

**Comment by kamleshchandnani** [comment]
Deploy preview for *leaf-ui* processing.

Built with commit 33606755dbe4ce5c17c42b831cdde9ce9e8b7de6

https://app.netlify.com/sites/leaf-ui/deploys/5ae6d5bfb312741d19f3e74f

File: src/Table/web/TBody.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import styled from 'styled-components';
+
+const TBody = styled.tbody`
```
Comment (srinivasdamam) [inline src/Table/web/TBody.js]: `Tbody` is much simpler.

File: src/Table/web/TD.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import theme from '../../theme';
+
+const TD = styled.td`
```
Comment (srinivasdamam) [inline src/Table/web/TD.js]: How about renaming to `Cell`?
`Table.Cell` makes more sense.

File: src/Table/web/TH.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import theme from '../../theme';
+
+const TH = styled.th`
```
Comment (srinivasdamam) [inline src/Table/web/TH.js]: `Head`

File: src/Table/web/Table.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Table.propTypes = {
+  color: PropTypes.oneOf(Object.keys(theme.color)),
+};
+
+Table.defaultProps = {
+  color: 'white',
+};
+
+Table.THead = THead;
```
Comment (srinivasdamam) [inline src/Table/web/Table.js]: Table.Head

File: src/Table/web/Table.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+Table.TBody = TBody;
```
Comment (srinivasdamam) [inline src/Table/web/Table.js]: Table.Body

File: src/Table/web/Table.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+Table.TH = TH;
+
+Table.TR = TR;
```
Comment (srinivasdamam) [inline src/Table/web/Table.js]: Table.Row

File: src/Table/web/Table.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+Table.TD = TD;
```
Comment (srinivasdamam) [inline src/Table/web/Table.js]: Table.Cell

File: src/Table/web/TD.js (hunk 2 above)
Comment (pragyajha) [reply src/Table/web/TD.js]: Makes sense! Was trying to keep naming in sync with html table tags.

File: src/Table/web/TH.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  padding: ${(props) => props.theme.px([2, 2.5])};
+  text-align: left;
```
Comment (srinivasdamam) [inline src/Table/web/TH.js]: can move `text-align` to props.

File: src/Table/web/Table.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+import theme from '../../theme';
+import THead from './THead';
+import TBody from './TBody';
+import TH from './TH';
+import TR from './TR';
+import TD from './TD';
+
+const Table = styled.table`
+  border-collapse: collapse;
+  background-color: ${(props) => props.color ? props.theme.color[props.color] : ''};
```
Comment (srinivasdamam) [inline src/Table/web/Table.js]: `props.color ? props.theme.color[props.color] : props.theme.color.white`

Comment (srinivasdamam) [inline src/Table/web/Table.js]: `props.theme.color[props.color] || props.theme.color.white`

File: src/Table/web/Table.js (hunk 7 above)
Comment (lakshyaranganath) [reply src/Table/web/Table.js]: Sticking to semantics, call this `Table.Data`

File: src/Table/web/Table.js (hunk 4 above)
Comment (lakshyaranganath) [reply src/Table/web/Table.js]: However, that doesn't mean that we're not okay with coming up with a new set of names that's at least close enough to the original html semantics.

File: src/Table/web/Table.story.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import { storiesOf } from '@storybook/react';
+import Card from '../../Card/web';
+import Select from '../../Select/web';
+import Tag from '../../Tag/web';
+import Text from '../../Text/web';
+import Spacer from '../../Spacer/web';
+import Table from './Table';
```
Comment (lakshyaranganath) [inline src/Table/web/Table.story.js]: `import Table from '../web';`

File: src/Table/web/Table.story.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+import makeStories from '../makeStories';
+
+makeStories(storiesOf, { Card, Table, Tag, Text, Select, Spacer });
```
Comment (lakshyaranganath) [inline src/Table/web/Table.story.js]: Go ahead and split this into multiple lines for better readability

**Pull request merged**